src/lib/OpenEXR/ImfInputFile.cpp

```diff
@@ -33,237 +33,657 @@
 
 #include <algorithm>
 #include <fstream>
+#include <iostream>
+#include <vector>
 
 OPENEXR_IMF_INTERNAL_NAMESPACE_SOURCE_ENTER
 
-using IMATH_NAMESPACE::Box2i;
-using IMATH_NAMESPACE::divp;
-using IMATH_NAMESPACE::modp;
+struct InputFile::Data
+{
+    static constexpr int kDefaultPart = 0;
 
-//
-// Struct InputFile::Data stores things that will be
-// needed between calls to readPixels
-//
+    Data (Context* ctxt, int t) : _ctxt (ctxt), _numThreads (t) {}
+    ~Data ()
+    {
+        deleteCachedBuffer ();
+    }
 
-struct InputFile::Data
+    const Header& getHeader (int part)
+    {
+        int pc = _ctxt->partCount ();
 #if ILMTHREAD_THREADING_ENABLED
-    : public std::mutex
+        std::lock_guard<std::mutex> lk (_mx);
 #endif
-{
-    Header header;
-    int    version;
-    bool   isTiled;
+        if (part < pc)
+        {
+            int cursz = static_cast<int> (_lazy_header_cache.size ());
+            if (part >= cursz)
+            {
+                _lazy_header_cache.resize (static_cast<size_t> (part) + 1);
```

Comment:
are you leaving room for a "safety" zero here?

Reply:
basically my question is whether this might induce extra pages, like say cache size is perfectly 4k, and adding a single 0 means it occupies two 4k pages instead of 1 now